connector/connector_github.go

```diff
@@ -0,0 +1,248 @@
+package connector
+
+import (
+	"encoding/json"
+	"fmt"
+	"html/template"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"path"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/coreos/dex/pkg/log"
+	"github.com/coreos/go-oidc/oauth2"
+	"github.com/coreos/go-oidc/oidc"
+)
+
+const (
+	GitHubConnectorType  = "github"
+	gitHubAuthorizeURL   = "https://github.com/login/oauth/authorize"
+	gitHubAccessTokenURL = "https://github.com/login/oauth/access_token"
+	gitHubAPIUserURL     = "https://api.github.com/user"
+)
+
+func init() {
+	RegisterConnectorConfigType(GitHubConnectorType, func() ConnectorConfig { return &GitHubConnectorConfig{} })
+}
+
+type GitHubConnectorConfig struct {
+	ID           string   `json:"id"`
+	ClientID     string   `json:"clientID"`
+	ClientSecret string   `json:"clientSecret"`
+	Scopes       []string `json:"scopes"`
```

Comment:
In case users of dex administrators wanted more than `user:email` in the future. Though admittedly I don't know what that would look like. I'll remove the option.